apps/challenges/views.py

```diff
@@ -777,6 +839,87 @@ def create_challenge_using_zip_file(request, challenge_host_team_pk):
         response_data = {"error": message}
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
+    # Manipulating the challenge data of the template challenge with data sent by hosts
+    if is_template_challenge:
+        yaml_file_data["title"] = challenge_data_from_hosts.get("title")
+        yaml_file_data["short_description"] = challenge_data_from_hosts.get(
+            "short_description"
```

Comment:
other than json fields can't you create a array with field names and loop through the array of fields to override value if its present in the request data